### PR TITLE
CNV_3044 Add Disk to VM

### PR DESCRIPTION
**Pull request by bgaydosrh**
Created new module for adding disks to VMs, conditionalized to reuse for both add to VM and VM templates.

Peer review requested from @aburdenthehand to ensure my conditionals and associated logic and syntax is valid.

Also content questions regarding GUI and associated functionality will be sent to Adam Litke and Tomas Jellinek.

Label **Peer Review Needed** and **enterprise-4.3**.

Many thanks!

Bob

### CODE REVIEW

**Comment by bgaydosrh** [comment]
I already discovered my comments re: inclusion in assemblies must be updated. Forgot that... will take care of it. Sorry.

File: modules/cnv-add-nic-to-vm.adoc [hunk 1]
```diff
@@ -0,0 +1,67 @@
+// Module included in the following assemblies:
```
Comment (aburdenthehand) [inline modules/cnv-add-nic-to-vm.adoc]: This file is added in #17826. Is it meant to be included in this PR as well?

Comment (bgaydosrh) [reply modules/cnv-add-nic-to-vm.adoc]: Hi @aburdenthehand,

I think I did a save to this file after a non-significant change and then added it. Probably not the best decision. Can I just delete the file from this PR in GitHub or should I do a new PR?

Thanks

Bob

Comment (bgaydosrh) [reply modules/cnv-add-nic-to-vm.adoc]: I guess I can just do a **git checkout --** on this file and create another commit, right? :-)

Comment (aburdenthehand) [reply modules/cnv-add-nic-to-vm.adoc]: @bgaydosrh - Yeah, that'll fix it.

Comment (bgaydosrh) [reply modules/cnv-add-nic-to-vm.adoc]: Thanks, did 
$ git checkout CNV_3044 -- modules/cnv-add-nic-to-vm.adoc
and
$ git rm modules/cnv-add-nic-to-vm.adoc

and it worked like a charm.

**Comment by bgaydosrh** [comment]
Code review complete.

@alexxa can you verify that the UI steps are the same whether using a template or not? Thanks. Once I have that, we can merge and pick to 4.3.

File: modules/cnv-add-disk-to-vm.adoc [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+[id="cnv-vm-add-disk_{context}"]
+
+= Adding a virtual disk to a {object}
+
+.Procedure
+
+. From the *{object-gui}* tab, select your {object}.
+. Select the *Disks* tab.
+. Click *Add Disks* to open the *Add Disk* window:
```
Comment (bmcelvee) [inline modules/cnv-add-disk-to-vm.adoc]: s/:/.

**Comment by bmcelvee** [comment]
One tiny nit, otherwise looks good to me!

File: modules/cnv-add-disk-to-vm.adoc (hunk 2 above)
Comment (bgaydosrh) [reply modules/cnv-add-disk-to-vm.adoc]: Fixed. Thank you. Should be ready for merge and pick now to 4.2 and 4.3

Comment (bgaydosrh) [reply modules/cnv-add-disk-to-vm.adoc]: Hi @bmcelvee -  we have found a solution to get the merge to 4.2. If you can comment out the Ifdefs and Ifevals from modules/cnv-add-disk-to-vm.adoc it should fix it. Are you up for that? I wasn't aware that 4.1/2.1 and 4.2/2.2 are tethered in content, so the conditional code is causing builder to barf as the referenced stuff is not in 4.1/2.1

Copying @aburdenthehand

Comment (aburdenthehand) [reply modules/cnv-add-disk-to-vm.adoc]: @bmcelvee @bgaydosrh - just to be clear, only commenting out the ifevals that relate to 
ifeval::["{context}" == "cnv-edit-vms"]

The second one (cnv-editing-vm-template) can stay, and is needed to provide the {object} variables.

Comment (bmcelvee) [reply modules/cnv-add-disk-to-vm.adoc]: Thanks! It's still not passing. I'll also tag Vikram to see if he might be able to provide some guidance.

Comment (aburdenthehand) [reply modules/cnv-add-disk-to-vm.adoc]: @bmcelvee - I noticed something else on that module and commented in your PR

**Pull request merged**